relay-sampling/src/config.rs

```diff
@@ -7,22 +7,36 @@ use serde::{Deserialize, Serialize};
 
 use relay_protocol::RuleCondition;
 
+/// Maximum supported version of dynamic sampling.
+///
+/// The version is an integer scalar, incremented by one on each new version:
+///  - 1: Initial version that uses `rules_v2`.
+///  - 2: Moves back to `rules` and adds support for `RuleConfigs` with string comparisons.
+const SAMPLING_CONFIG_VERSION: u16 = 2;
+
 /// Represents the dynamic sampling configuration available to a project.
 ///
 /// Note: This comes from the organization data
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct SamplingConfig {
+    /// The required version to run dynamic sampling.
+    ///
+    /// Defaults to legacy version (`1`) when missing.
+    #[serde(default = "SamplingConfig::legacy_version")]
+    pub version: u16,
+
     /// The ordered sampling rules for the project.
     ///
     /// This field will remain here to serve only for old customer Relays to which we will
     /// forward the sampling config. The idea is that those Relays will get the old rules as
     /// empty array, which will result in them not sampling and forwarding sampling decisions to
     /// upstream Relays.
-    #[serde(default, skip_deserializing)]
+    #[serde(default)]
     pub rules: Vec<SamplingRule>,
 
     /// The ordered sampling rules v2 for the project.
```

Comment:
Should note that is only used for version 1, even though it has v2 in its name

```suggestion
    /// Ordered sampling rules, used in version 1. Removed in favour of `Self::rules` in version 2.
```